Analyzer/test/MyAnalysis.C

```diff
@@ -68,9 +96,11 @@ template<typename Analyze> void run(std::set<AnaSamples::FileSummary> vvf,
         tr.registerFunction( std::move(runFisher) );
         tr.registerFunction( std::move(bjet) );
         tr.registerFunction( std::move(commonVariables) );
+        tr.registerFunction( std::move(scaleFactors) );
         tr.registerFunction( std::move(baseline) );
 
         // Loop over all of the events and fill histos
+        a.InitHistos(tr);
```

Comment:
Can we not InitHistos like this? Can you do it the same way that I do it in my analyzers:
https://github.com/StealthStop/Analyzer/blob/master/Analyzer/src/Analyze0Lep.cc#L384-L389

Reply:
In general I'm not sure we should or need to pass in the nTupleReader to the InitHistos like this. I think we had decided not to change the names of histogram files?

Reply:
I think how I do cuts and pass in the information for Analyze0Lep is better and will save us more time in the future. That way we are not all having to have the same type of InitHistos function.

Reply:
In general, I guess we did not need to pass the nTupleReader into the InitHistos. It was important for my B Tag Efficiency analyzer because it had to read in the sample set and sample collection in order to properly calculate the right weights needed for the b tag efficiency. I thought it might be easier to just make the change across the board.

I can change it back if that's what you want, and if you think doing InitHistos your way is better, I can adapt accordingly.

Reply:
I think it is better because we can have different InitHistos for each Analyzer and not have it in MyAnalyzer.C plus if you want you can pass in stuff like my cut map to make your life easier.

Reply:
Sounds good. I'm currently going through each analyzer and making sure it compiles.